test/mechanics/integrands/HistoryData.cpp

```diff
@@ -0,0 +1,397 @@
+    //! @param delta_t: Time increment
+    void UpdateHistoryData(const NuTo::CellData& cellData, const NuTo::CellIpData& cellIpData, DofType dofType,
+                           double delta_t)
+    {
+        // Get history data
+        auto& hisData = GetIPHistoryData(cellData.GetCellId(), cellIpData.GetIPNum());
+
+        // Calculate necessary values for update
+        NuTo::BMatrixStrain B = cellIpData.GetBMatrixStrain(dofType);
+        NuTo::NodeValues u = cellData.GetNodeValues(dofType);
+        EngineeringStrainPDE<1> deltaCreep{DeltaCreep(hisData, delta_t)};
+        NuTo::EngineeringStrainPDE<1> strain = B * u;
+        NuTo::EngineeringStressPDE<1> stress = Stress(strain, delta_t, cellData.GetCellId(), cellIpData.GetIPNum());
+        MechanicsTangent E = Tangent(strain, delta_t, cellData.GetCellId(), cellIpData.GetIPNum());
+        NuTo::EngineeringStrainPDE<1> deltaStrain = strain - hisData.prevStrain;
+
+        // The actual update
+        hisData.prevStrain = strain;
+        hisData.prevStress = stress;
+        for (unsigned int i = 0; i < mE_KC.rows(); ++i)
+            hisData.gamma[i] = Lambda(delta_t, i) * E / mE_KC[i] * (deltaStrain - deltaCreep) +
+                               Beta(delta_t, i) * hisData.gamma[i];
+    }
+
+private:
+    //! @brief Calculates the algorithm specific parameter beta
+    //! @param delta_t: Time increment
+    //! @param index: Index of the Kelvin Unit
+    //! @return Algorithm specific parameter beta
+    double Beta(double delta_t, unsigned int index) const
+    {
+        assert(index < mT_KC.rows());
+        return std::exp(-delta_t / mT_KC[index]);
+    }
+
+    //! @brief Calculates the algorithm specific parameter lambda
+    //! @param delta_t: Time increment
+    //! @param index: Index of the Kelvin Unit
+    //! @return Algorithm specific parameter lambda
+    double Lambda(double delta_t, unsigned int index) const
+    {
+        assert(index < mT_KC.rows());
+        return mT_KC[index] / delta_t * (1 - Beta(delta_t, index));
+    }
+
+    //! @brief Calculates the creep strain increment
+    //! @param hisData: History data object
+    //! @param delta_t: Time increment
+    //! @return Creep strain increment
+    EngineeringStrainPDE<1> DeltaCreep(const CreepHistoryData& hisData, double delta_t) const
+    {
+        EngineeringStrainPDE<1> deltaCreep;
+        deltaCreep[0] = 0.;
+        for (unsigned int i = 0; i < mE_KC.rows(); ++i)
+            deltaCreep[0] += (1. - Beta(delta_t, i)) * hisData.gamma[i];
+        return deltaCreep;
+    }
+};
+
+
+// %%% Mesh generation %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+
+MeshFem Mesh1D()
+{
+    MeshFem mesh;
+    const InterpolationSimple& interpolation = mesh.CreateInterpolation(InterpolationTrussLinear(1));
+    NodeSimple* nr = nullptr;
+    constexpr unsigned int numNodes = numElements + 1;
+    constexpr double elementLength = SpecimenLength / numElements;
```

Comment:
I would be nice if you could move this to `UnitMeshFem::CreateTrusses(int numElements)` for everyone to access.

Reply:
Can do that.